husarion_ugv_crsf_teleop/husarion_ugv_crsf_teleop/crsf/parser.py

```diff
@@ -0,0 +1,133 @@
+# Based on https://github.com/crsf-wg/crsf/wiki/Python-Parser by crsf-wg
+
+from typing import Tuple, Callable
+from enum import Enum
+from .message import CRSFMessage, PacketType, CRSF_SYNC, CRSF_SYNC_EDGETX, unpack_channels
+
+class CRSFParser:
+    class State(Enum):
+        SEEK_SYNC = 0,
+        MSG_LEN = 1,
+        MSG_TYPE = 2,
+        MSG_EXT_DEST = 3,
+        MSG_EXT_SRC = 4,
+        MSG_PAYLOAD = 5,
+        MSG_CRC = 6
+        
+    class Result(Enum):
+        PACKET_VALID = 0,
+        IN_PROGRESS = 1,
+        PACKET_INVALID = 2
+        
+    _msg: CRSFMessage
+    _buffer: bytearray = bytearray()
+    state: State = State.SEEK_SYNC
+    on_message: Callable[[CRSFMessage], None]
+    
+    def __init__(self):
+        pass
+    
+    # Parses CRSF messages from entire buffer
+    def parse(self, data: bytearray):
```

Comment:
```suggestion
    def parse(self, entire_buffer: bytearray):
```
This comment can be removed with different name of the variable.

Reply:
Dropping the comment, imo parse(data) should carry enough context